plugins/module_utils/foundation/image_nodes.py

```diff
@@ -109,10 +141,11 @@ def _build_spec_hypervisor_iso(self, payload, value):
             hi_val = value[hi]
             hi_name = self._ahv2kvm(hi_name)
             hypervisor_iso[hi_name] = {}
-            if "checksum" in hi_val:
-                hypervisor_iso[hi_name]["checksum"] = hi_val.get("checksum")
-            if "filename" in hi_val:
-                hypervisor_iso[hi_name]["filename"] = hi_val.get("filename")
+            if "checksum" not in hi_val or "filename" not in hi_val:
+                return None, "checksum and filename are required parameter for hypervisor iso" 
```

Comment:
Change the error as
"Missing params: checksum, filename. Mandatory for hypervisor iso"

Infact add the ansible required_if() condition in module_spec. So that it can be validated at the start.

Reply:
Removed validations from here as they are required in module spec